Assets/Tests/InputSystem/SwitchTests.cs

```diff
@@ -80,6 +80,61 @@ private static SwitchProControllerHIDInputState StateWithButton(SwitchProControl
         }.WithButton(button);
     }
 
+    [Test]
+    [Category("Devices")]
+    public void Devices_SwitchProAxisJitter_DoesntMakeDeviceCurrent()
+    {
+        var device1 = InputSystem.AddDevice<SwitchProControllerHID>();
+        var device2 = InputSystem.AddDevice<SwitchProControllerHID>();
+
+        InputSystem.QueueStateEvent(device1,
+            new SwitchProControllerHIDInputState
+            {
+                leftStickX = 0x7f,
+                leftStickY = 0x7f,
+                rightStickX = 0x7f,
+                rightStickY = 0x7f,
+            });
+        InputSystem.QueueStateEvent(device2,
+            new SwitchProControllerHIDInputState
+            {
+                leftStickX = 0x7f,
+                leftStickY = 0x7f,
+                rightStickX = 0x7f,
+                rightStickY = 0x7f,
+            });
+        InputSystem.Update();
+        Assert.That(Gamepad.current, Is.EqualTo(device2));
+
+        var jitterMask = 0b111;
+
+        InputSystem.QueueStateEvent(device1,
+            new SwitchProControllerHIDInputState
+            {
+                leftStickX = (byte)(0x7f - jitterMask),
+                leftStickY = (byte)(0x7f - jitterMask),
+                rightStickX = (byte)(0x7f - jitterMask),
+                rightStickY = (byte)(0x7f - jitterMask),
+            });
+        InputSystem.Update();
+        Assert.That(Gamepad.current, Is.EqualTo(device2));
+
+        InputSystem.QueueStateEvent(device1,
+            new SwitchProControllerHIDInputState
+            {
+                leftStickX = (byte)(0x7f - jitterMask - 1),
+                leftStickY = (byte)(0x7f - jitterMask - 1),
+                rightStickX = (byte)(0x7e - jitterMask - 1),
+                rightStickY = (byte)(0x7e - jitterMask - 1),
+            });
+        InputSystem.Update();
+        Assert.That(Gamepad.current, Is.EqualTo(device1));
+
+        ////TODO: 7f <-> 80 jitter does break this, because it changes all the bits:
```

Comment:
Based on the InputControl configuration at the bottom of this PR it seems like the jitterMask is set to 0x7. Wouldn't this imply that this behavior you mention related to comparing binary rather that than numerical difference imply that such a  "false positive" would occur for a "staircase model" for any single bit change outside the jitter mask e.g. 0x7 to 0x8 (and the opposite), as well 0xf to 0x10 (and the opposite) etc, having certain points along the value axis that will trigger this when transitioning over a multiple of 8? 

I think an interesting test case for this is sweeping a "representative value" e.g. x by incrementing it by 1 in a loop and have a value y = x + error where error is coming from a e.g. gaussian distribution or at least random distribution which never exceeds e.g. the jitterMask value range. It might be that the starcaising here is not possible to solve without LPF as you mention so maybe such a test case would only fail? A test case with known seend random generators can be useful though when testing models trying to mitigate or manage noise to create invariant tests where e.g. seed would typically be test parameter into the test. It seems like this test case is verifying a fixed specific case but it might be desirable to complement with a test case utilizing a psuedorandom element. Especially if implementation changes over time such a test case would still be valid.